cpu/esp32/irq_arch.c

```diff
@@ -19,58 +19,101 @@
  */
 
 #include "irq_arch.h"
+
+#include "esp_attr.h"
 #include "esp_err.h"
-#include "esp32/rom/ets_sys.h"
-#include "soc/dport_reg.h"
-#include "xtensa/xtensa_api.h"
+#include "freertos/FreeRTOS.h"
+#include "hal/interrupt_controller_types.h"
+#include "hal/interrupt_controller_ll.h"
+#include "rom/ets_sys.h"
+#include "soc/periph_defs.h"
+#include "esp_intr_alloc.h"
 
 #define ENABLE_DEBUG 0
 #include "debug.h"
 
-struct _irq_alloc_table_t {
+#ifndef ETS_CAN_INTR_SOURCE
+#define ETS_CAN_INTR_SOURCE     ETS_TWAI_INTR_SOURCE
+#endif
+
+typedef struct intr_handle_data_t {
     int      src;   /* peripheral interrupt source */
     uint32_t intr;  /* interrupt number */
+    uint8_t  level;
+} intr_handle_data_t;
+
+/* TODO change to a clearer approach */
+static struct intr_handle_data_t _irq_data_table[] = {
+    { ETS_FROM_CPU_INTR0_SOURCE, CPU_INUM_SOFTWARE, 1 },
+    { ETS_TG0_WDT_LEVEL_INTR_SOURCE, CPU_INUM_WDT, 1 },
+    { ETS_TG0_T0_LEVEL_INTR_SOURCE, CPU_INUM_RTT, 1 },
+#if defined(MCU_ESP32) || defined(MCU_ESP32S2) || defined(MCU_ESP32S3)
```

Comment:
Would this work?

```suggestion
#if defined(ETS_TG0_T1_LEVEL_INTR_SOURCE)
```

Reply:
I don't think so. For ESP32, it is a define, but for ESP32-S2 and ESP32-S3 it is an enumeration.

Even though all these defines are not necessary here for ESP32 and could be added later, but when I went through the ESP-IDF to gather the information what is the same for all SoCs and what is different, I just wrote it down. Otherwise, I would have to check it again and again.